### ISSUE TITLE
Asserterror:'dict' object has no attribute load_state_dict

### ISSUE THREAD

**Issue by SiYue1252** [ultralytics/yolov5#12905]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

My code:
model = torch.load('yolov5_path/best.pt',map_location=device)
model.load_state_dict(torch.load('yolov5_path/state.pth',map_location=device))
model.eval()
Asserterror:'dict' object has no attribute load_state_dict
(state.pth came from :
import torch
m = torch.load("path/best.pt")
torch.save(m["model"].state_dict(), "path/state.pth")

**Comment by glenn-jocher** [comment]
Hello! 😊 It appears there's a small misunderstanding in how the model is loaded from the `.pt` file. `torch.load('yolov5_path/best.pt')` loads a pre-trained YOLOv5 model as a dictionary, including model weights and possibly other information, rather than directly as a model object. That's why you're seeing an error when trying to use `.load_state_dict()` on a dictionary. Here's a quick fix:

```python
# Load the model
model = torch.load('yolov5_path/best.pt', map_location=device)['model']
# Now you can directly use the model for inference without loading state_dict again
model.eval()
```

This loads your model and sets it to evaluation mode, ready for inference. Remember, the `torch.load('yolov5_path/best.pt')` command returns the entire checkpoint, from which you only need the `model` for inference or further training tasks. No need to load `state.pth` again if you've already got everything in `best.pt`. Hope this clears things up!
